linode/manager
# POC: [M3-7147] - Accessibility: ConfirmationDialog form submission via `enter` key

## Description 📝
This PR adds keyboard support to the `ConfirmationDialog` component. It allows a user navigating Cloud Manager via keyboard to submit the confirmation modal via pressing `enter/return` while the tabulation is on the Dialog itself.

This feature is available for native dialogs containing a form, alas MUI dialogs don't have that capability out of the box.
This contribution surfaces that functionality back and adds better semantic markups to the Dialog forms.

I expect this contribution to generate conversations on how we want to handle this globally. For instance, in this PR I narrowed the ability to submit the form via `enter` when user is focused on the modal. Which means while focused on a different element (ex: a `<select />` within the form), pressing enter won't submit the form (this differs from native behavior! - _which we can decide to adopt or not_).

The difficulty lies in deciding **when**  we want this behavior, and when we don't want it, and if we want to mimic native behavior. If so, any form a user is currently filling could be submitted For instance, the Linode Create form (and in general forms with few or more fields) don't seem to be great candidates. 

While it is our goal to improve form keyboard handling through the app, the `ConfirmationModal` is a perfect candidate to start showcasing the behavior and will hopefully inspire expanding this accessibility feature to the rest of our forms.

> [!WARNING]
> This PR only addresses the `ConfirmationModal` component, therefore this improvement will not apply to modals that are built without the component (ex: the rebuild linode dialog)

## Changes  🔄
- Allow submitting `ConfirmationDialog` forms via pressing `enter` while focused
- Improve dialog form semantic markup for container and submit button
- Add new shiny ✨  CY component test

## Preview 📷

https://github.com/user-attachments/assets/223c99bc-6bbd-4b0b-9fa0-bf6360e179fd

## How to test 🧪

### Verification steps
- Test various Confirmation Modals (ex: Reboot Linode, Delete Linode etc)
- Run the component test locally (it is not in the pipeline yet, but VERY soon - see https://github.com/linode/manager/pull/10926)
  - run `yarn cy:debug`
  - choose component testing
  - run `confirmation-dialog.spec.tsx` 

## As an Author I have considered 🤔

*Check all that apply*
